pennylane_lightning/src/gates/DynamicDispatcher.hpp

```diff
@@ -134,6 +134,7 @@ template <typename PrecisionT> class DynamicDispatcher {
         -> std::vector<Gates::KernelType> {
         std::vector<Gates::KernelType> kernels;
 
+        kernels.reserve(kernel_names_.size());
```

Comment:
CT16 complained about this?

Reply:
Yeap. If the size of the vector is known prior, CT16 requires reserving the memory first as it is performant.